gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/graph.ts

```diff
@@ -119,6 +134,13 @@ export class VertexProperty<T1 = any, T2 = any> extends Element {
   }
 }
 
+export type VertexProperties<
+  TLabel extends string = string,
+  TValue = any,
+  TProperties extends Record<string, any> = Record<string, any>,
+  TId = any,
+> = [VertexProperty<TLabel, TValue, TProperties, TId>, ...Array<VertexProperty<TLabel, TValue, TProperties, TId>>];
```

Comment:
Sorry, but I don't know this notation and it's a bit hard to find any information on it as I don't even know what to search for here 😅 
Would you be so kind to provide me with some documentation for this kind of type definition or maybe the name of the concept for this so I can search for it myself? :-)

Reply:
I believe you are referring to this part?

```ts
[VertexProperty<TLabel, TValue, TProperties, TId>, ...Array<VertexProperty<TLabel, TValue, TProperties, TId>>]
```

It basically will ensure the array will have at least 1 element.

You can find more information regarding it [here](https://www.typescriptlang.org/docs/handbook/variable-declarations.html#spread) :D

Reply:
Interesting, didn't know the spread operator. Thanks for the explanation :)